### PR TITLE
ARROW-1949: [Python/C++] Add option to Array.from_pandas and pyarrow.array to perform unsafe casts

### PR DESCRIPTION
**Pull request by kszucs**

### CODE REVIEW

File: python/pyarrow/array.pxi [hunk 1]
```diff
@@ -17,8 +17,8 @@
 
 
 cdef _sequence_to_array(object sequence, object mask, object size,
-                        DataType type,
-                        CMemoryPool* pool, c_bool from_pandas):
+                        DataType type, CMemoryPool* pool, c_bool from_pandas,
+                        c_bool safe):
```
Comment (kszucs) [inline python/pyarrow/array.pxi]: TODO: Implementation is missing from here

Comment (kszucs) [reply python/pyarrow/array.pxi]: ConvertPySequence [assumes](https://github.com/apache/arrow/blob/master/cpp/src/arrow/python/python_to_arrow.cc#L926) strict conversion if datatype is explicitly passed.

**Review by xhochy — APPROVED** [review]
+1, LGTM

**Comment by kszucs** [comment]
Thanks Uwe!

**Pull request closed without merging**